ocs_ci/ocs/resources/cloud_manager.py

```
@@ -756,3 +768,53 @@ def __init__(
+
+class AzureSTSClient(AzureClient):
+    """
+    Implementation of an Azure Client for STS (managed identity) clusters.
+
+    Reuses AzureClient's blob_service_client for ULS operations but creates
+    a different k8s secret with managed identity fields instead of AccountKey.
+
+    """
+
+    @config.run_with_provider_context_if_available
+    def __init__(self, full_auth_dict, *args, **kwargs):
```

Ruff (0.15.15)

[warning] Missing return type annotation for special method `__init__`

Add return type annotation: `None`

(ANN204)